crates/generator/src/generator.rs

```diff
@@ -185,6 +193,7 @@ impl Generator {
                     return Err(err.into());
                 }
             }
+            self.redir_log_handler.flush(exit_code);
```

Comment:
If the function returns early, `flush` won't be called.

Also, could `machine_run` ever run concurrently?

Reply:
`machine_run` never runs concurrently. 
It's OK if this fn returns early. I think that only happens when the process exits. Plus, log printing is not that critical.

Reply:
If `flush` is not called, the next run will have logs of this run. A workaround is to clear the sink in setup instead flush.